server/src/main/java/pl/edu/agh/RaftServer.java

```diff
@@ -49,23 +52,28 @@
     private static final ScheduledExecutorService TIMEOUT_EXECUTOR = Executors.newSingleThreadScheduledExecutor();
 
     private final SocketAddress localAddress;
-    private final TcpServer<ByteBuf, ByteBuf> tcpServer;
-    private final Map<SocketAddress, Connection<ByteBuf, ByteBuf>> serverConnections;
+    private Map<SocketAddress, Connection<ByteBuf, ByteBuf>> serverConnections = Maps.newHashMap();
+    private Connection<ByteBuf, ByteBuf> clientConnection;
 
     private State state = State.FOLLOWER;
     private int currentTerm = 0;
     private AtomicInteger votesCount = new AtomicInteger(0);
     private SocketAddress votedFor;
     private ScheduledFuture timeout;
 
+    private volatile Map<String, Integer> keyValueStore = Maps.newHashMap();
```

Comment:
I'm not sure if the volatile keyword is necessary here. There are no multiple read/writes to that variable, we have only one leader and key-value store operations are commited synchronously. Correct me if I'm wrong.
Please do it like that:
```java
private final Map<String, Integer> keyValueStore = new HashMap<>();
```